File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/filter/primitive/FloatFilterParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.primitive;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.FloatFilter;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.FilterParser;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.finder.Finder;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.Validator;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.impl.FloatValidator;

@Component
public class FloatFilterParser implements FilterParser<FloatFilter, Float> {

    @Override
    public Validator<Float> createChecker(FloatFilter annotation, Finder finder) {
        return new FloatValidator(annotation.lessThan(),
                                  annotation.lessOrEqual(),
                                  annotation.moreThan(),
                                  annotation.moreOrEqual(),
                                  annotation.equal());
    }

    @Override
    public Class<FloatFilter> getAnnotationClass() {
        return FloatFilter.class;
    }

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/controller/annotations/parser/filter/primitive/FloatFilterParser.kt
package io.github.oleksivio.tl.kbot.core.controller.annotations.parser.filter.primitive

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.FloatFilter
import io.github.oleksivio.tl.kbot.core.controller.annotations.parser.filter.FilterParser
import io.github.oleksivio.tl.kbot.core.controller.annotations.parser.finder.Finder
import io.github.oleksivio.tl.kbot.core.controller.handler.check.Validator
import io.github.oleksivio.tl.kbot.core.controller.handler.check.impl.FloatValidator

class FloatFilterParser :
    FilterParser<FloatFilter, Float> {

    override val parserAnnotationClass = FloatFilter::class

    override fun createChecker(annotation: FloatFilter, finder: Finder): Validator<Float> {
        return FloatValidator(
            annotation.lessThan,
            annotation.lessOrEqual,
            annotation.greatThan,
            annotation.greatOrEqual,
            annotation.equal
        )
    }
}
